src/main.rs

```diff
@@ -59,7 +59,7 @@ pub fn main() {
 }
 
 #[cfg_attr(tarpaulin, skip)]
-pub fn run_opt(opt: &Opt) -> Result<bool, Error> {
+pub fn run_opt(opt: &Opt) -> Result<(bool, Option<SvData>), Error> {
```

Comment:
The return type looks redundant here.
I think it should now be `Result<Option<SvData>, Error>` because there's no case where `all_pass` is `false` and `ret` is `Some(svdata)`, and no case where `all_pass` is `true` and `ret` is `None`.

Reply:
I totally agree with that. It will become cleaner and with no redundancy. I'm on it to fix it.